### PR TITLE
Field layout preview component trigger

### PR DESCRIPTION
**Pull request by fredrikkarlsson-pervanovo**
This trigger is needed to be able to write addons that generates layout component previews for custom components

### CODE REVIEW

File: modules/Cockpit/assets/components/field-layout.tag [hunk 1]
```diff
@@ -522,7 +522,9 @@
         }
 
         getPreview(component) {
-            //console.log(component)
+            var triggerOutput = [];
+            App.trigger('field.layout.component.preview', {component: component, def: this.components[component.component], output: triggerOutput});
+            if (triggerOutput.length > 0) return triggerOutput.join('');
```
Comment (aheinze) [inline modules/Cockpit/assets/components/field-layout.tag:527]: Why an array?

Comment (fredrikkarlsson-pervanovo) [reply modules/Cockpit/assets/components/field-layout.tag:527]: For the handler to be able to send back data it has to be some kind of referenceable data type, ie. object or array. A string would not be able to be changed from inside the handler.

Here is some code of the handler in our addon that uses this trigger:
```
 App.on("field.layout.component.preview", function (trigger) {
        var componentName = trigger.params.component.component;
        var value = trigger.params.component.settings;
        var output = trigger.params.output;
        switch (componentName) {
            case "custom_heading":
                output.push(badge(value.tag) + value.text);
                return;
```

Comment (fredrikkarlsson-pervanovo) [reply modules/Cockpit/assets/components/field-layout.tag:527]: @aheinze Any toughts on this? I just rebased the PR to stay in sync.

**Comment by fredrikkarlsson-pervanovo** [comment]
Here is an addon I just published that needs this PR: https://github.com/Pervanovo/CockpitLayoutComponentPreviews

File: modules/Cockpit/assets/components/field-layout.tag (hunk 1 above)
Comment (aheinze) [reply modules/Cockpit/assets/components/field-layout.tag:527]:
```js
var output = '',
    params = {
        component: component, 
        def: this.components[component.component],
        render: function(contents) {
            output = contents || '';
        }
    };

App.trigger('field.layout.component.preview', params);

if (output) return output;
```

your handler would look then like:

```js
 App.on("field.layout.component.preview", function (trigger) {
        var componentName = trigger.params.component.component;
        var value = trigger.params.component.settings;
        var render = trigger.params.render;
        switch (componentName) {
            case "custom_heading":
                render(badge(value.tag) + value.text);
                return;
```

Comment (fredrikkarlsson-pervanovo) [reply modules/Cockpit/assets/components/field-layout.tag:527]: Great, I will rewrite it!

**Comment by fredrikkarlsson-pervanovo** [comment]
@aheinze Changed according to your feedback!

**Comment by aheinze** [comment]
> @aheinze Changed according to your feedback!
I think you still need to compile it (re-create components.js)

**Comment by fredrikkarlsson-pervanovo** [comment]
> > @aheinze Changed according to your feedback!
> 
> I think you still need to compile it (re-create components.js)

Silly me, I forgot... Done!

**Comment by aheinze** [comment]
Thanks!

**Pull request merged**